src/styles/calendar.scss

```diff
@@ -0,0 +1,48 @@
+body {
+  margin: 0;
+  font-family: Arial, sans-serif;
+}
+
+@each $day, $n in $days-of-week {
+  .calendar--start-#{$day} .calendar__day:first-child {
+    margin-left: ($calendar-day-width + (3 * $day-gap)) * $n;
+  }
+}
+
+@for $days-count from 28 through 31 {
+  .calendar--length-#{$days-count} .calendar__day:nth-child(n + #{$days-count + 1}) {
+    display: none;
+  }
+}
+
+.calendar {
+  box-sizing: border-box;
+  display: flex;
+  gap: $day-gap;
+  flex-wrap: wrap;
+  padding: 10px;
+  max-width: calc($calendar-day-width * 8);
+
+  &__day {
+    width: $calendar-day-width;
+    height: $calendar-day-height;
+    background-color: #eee;
+    border: 1px solid black;
```

Comment:
The border should be included in the box-sizing of the elements. Add 'box-sizing: border-box;' to .calendar__day.